keps/sig-scheduling/5278-nominated-node-name-for-expectation/README.md

```diff
@@ -0,0 +1,802 @@
+and doesn't have the implementation change from this KEP,
+nothing goes wrong because kube-apiserver just clears `NominatedNodeName` from the pods at the binding API,
+which is fine by the today's scheduler implementation as well.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: NominatedNodeNameForExpectation
+  - Components depending on the feature gate: kube-scheduler, kube-apiserver
+
+###### Does enabling the feature change any default behavior?
+
+Pods that are processed by Permit or PreBind plugins get NominatedNodeName during binding cycles.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes.
+The feature can be disabled in Beta version by restarting the kube-scheduler and kube-apiserver with the feature-gate off.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The scheduler just again starts to put NominatedNodeName at the beginning of binding cycles (if applicable). 
+
+###### Are there any tests for feature enablement/disablement?
+
+No.
+This feature is only changing when a NominatedNodeName field will be set - it doesn't introduce a new API. 
+However reacting to it is purely in-memory, so enablement/disablement tests wouldn't really differ from regular feature tests.
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+The scheduler and kube-apiserver are involved in this feature.
+
+If upgrading the scheduler fails somehow, new Pods won't be scheduled anymore until rolling back,
+while Pods, which are already scheduled, won't be affected.
+If upgrading kube-apiserver fails somehow, the whole Kubernetes will not be able to function properly until rolling back.
+
+Even if one of them cannot be upgraded properly somehow, and gets rolled back,
+there'll be nothing behaving wrong in the scheduling flow, see [Version Skew Strategy](#version-skew-strategy).
+
+###### What specific metrics should inform a rollback?
+
+- The `schedule_attempts_total` metric with the `error` label is increasing abnormally.
+- The `scheduler_pod_scheduling_sli_duration_seconds` or `scheduling_attempt_duration_seconds` gets too long.
+  - Although, for pods that have to go through Permit/PreBind plugins, it's expected that their scheduling+binding latency would get higher because of an additional API call for NominatedNodeName.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+We will do the following manual test after implementing the feature:
+
+1. upgrade
+2. set NNN to non-existing node
+3. ensure it won't get cleared
+4. downgrade
+5. ensure that it gets cleared
+6. upgrade
+7. set NNN to non-existing node
+8. ensure it isn't cleared again
```

Comment:
Normally, this is required for beta graduation, but given this is starting with beta it's hard to do the test before the implementation. But please once you land the code change to update this fragment of the document with the steps performed and results, and ping me so I'm aware this has happened.